lib/modules/showImages.js

```diff
@@ -583,22 +586,24 @@ export function toggleThingExpandos(thing, scrollOnExpando) {
 	}
 }
 
-export function mediaBrowseMode(oldThing, newThing) {
-	if (autoExpandActive) return false;
+function mediaBrowse(selected, unselected, options) {
+	if (!selected || !options.mediaBrowse || autoExpandActive) return;
 
-	const oldExpando = oldThing && oldThing.getEntryExpando();
-	const newExpando = newThing && newThing.getEntryExpando();
+	const oldExpando = unselected && unselected.getEntryExpando();
+	const newExpando = selected && selected.getEntryExpando();
 
 	if (oldExpando) {
 		mediaBrowseModeActive = oldExpando.expandWanted || oldExpando.open;
 	}
 
 	if (mediaBrowseModeActive) {
 		if (oldExpando) oldExpando.collapse();
-		if (newExpando) newExpando.expand();
-	}
 
-	return newExpando && mediaBrowseModeActive;
+		if (newExpando) {
+			newExpando.expand();
+			options.didAutoExpand = true;
```

Comment:
I think I'd prefer moving this entire function to keyboardNav (but still separated from `move`), so you can replace this with `if (module.options.scrollOnExpando.value) options.scrollStyle = 'top';` instead of setting `didAutoExpand` here and reading it in a getter in another file, which is a bit too hacky IMO.